### PR TITLE
Absolute files paths

### PR DESCRIPTION
**Pull request by AxGord**

### CODE REVIEW

File: src/Main.hx [hunk 1]
```diff
@@ -147,7 +147,7 @@ class Main {
 				return workspaceFolder.uri.fsPath;
 			}
 		}
-		return null;
+		return Vscode.workspace.workspaceFolders[0].uri.fsPath; // try first workspace folder
```
Comment (AlexHaxe) [inline src/Main.hx]: checkstyle should only happen for files within your workspace, so `null` is ok or even a requirement for files that were opened from outside (e.g. when you follow a type or method into your haxelib folder)

**Comment by AlexHaxe** [comment]
Can you add a little description of the problem you are trying to solve, because I'm not sure I understand what you are trying to achieve.

**Comment by AxGord** [comment]
Its for haxe dependencies panel. I have some dev libs connected to project and I want check style, because i'm edit code. The restriction of the workspace prevents this.

**Comment by AlexHaxe** [comment]
I've added a new parameter called `haxecheckstyle.externalSourceRoots` to allow additional root folders outside of your workspace.
Please test if it works for you / solves your problem.

**Comment by AxGord** [comment]
Works with commit e2fe325

**Comment by AlexHaxe** [comment]
I don't think that extra code in `fileInSourcePaths` is necessary. 
`haxecheckstyle.externalSourceRoots` settings is meant to be used in addition to `haxecheckstyle.sourceFolders`, not as a replacement. Of course this assumes that your dependencies have a folder structure that contains a `src` or `Source` folder (I think most of them do). 
So it should work without your modification

**Comment by AxGord** [comment]
Without modification does not work. Changes in fileInSourcePaths needed. Otherwise, the function fileInSourcePaths will return false.

**Comment by AxGord** [comment]
var rootPath = normalizePath(Path.join([rootFolder, path]));
rootFolder => "/path/to/lib"
path => "/path/to/lib"
rootPath => "/path/to/lib/path/to/lib" - wrong path

**Comment by AlexHaxe** [comment]
If `haxecheckstyle.externalSourceRoots` contains `["/usr/share/haxelib/formatter/1,3,0", "/usr/share/haxelib/checkstyle/2,4,2"]` and `haxecheckstyle.sourceFolders` is `["src"]`, then all files in `<workspaceRoot>/src`, `/usr/share/haxelib/formatter/1,3,0/src` and `/usr/share/haxelib/checkstyle/2,4,2/src` should match in `fileInSourcePaths`.

You shouldn't use full paths in `haxecheckstyle.sourceFolders`, it's meant as a collection of folder names where Haxe source files sit relative to their workspace roots. `haxecheckstyle.externalSourceRoots` only adds workspace roots that are outside of your current VSCode workspace.

I know that `haxecheckstyle.sourceFolders = ["src"]` will make every `src` folder in every workspace root available to checkstyling, but that shouldn't be an issue. You can always disable / exclude files from checkstyle, if you have one `src` folder where checkstyle shouldn't run.

**Comment by AxGord** [comment]
Thanks, I changed the configuration, now everything works.

**Pull request merged**